assets/src/edit-story/components/carousel/carouselProvider.js

```diff
@@ -38,10 +39,37 @@ function CarouselProvider({ availableSpace, children }) {
   );
 
   const [listElement, setListElement] = useState(null);
+  const [previewImages, setPreviewImages] = useState(null);
   const pageRefs = useRef([]);
+  const previewRefs = useRef(null);
 
   const numPages = pages.length;
 
+  useLayoutEffect(() => {
+    if (pages && pageRefs.current?.[pages[0]?.id] && !previewRefs.current) {
+      pages.forEach(({ id }) => {
+        const element = pageRefs.current[id];
+        const preview = element.querySelector('button');
+        htmlToImage
+          .toJpeg(preview, { quality: 0.75 })
```

Comment:
Note: tried out both `toSvg` and `toJpeg` here. It looks like `toJpeg` takes more time to create the initial images but the rendering is faster. Since perhaps the initial images creation can be done in a "lazy" way, then faster rendering might be preferred.